domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/TimelineEvent.kt

```diff
@@ -42,25 +42,33 @@ data class TimelineEvent(
    * The date and time when the event occurred.
    */
   val timestamp: Instant,
+  /**
+   * An optional correlationId for this and any other TimelineEvents that occurred at the same time (i.e. within same
+   * atomic action). For example, this could be an update to a Goal and one or more of its child Steps.
+   */
+  val correlationId: UUID?,
```

Comment:
Interesting that its optional .... I'm sure I'll discover why as I see the rest of the change in this PR, but I'm wondering when we might ever create a timeline event that does not have a correlation id .... 🤔